### ISSUE TITLE
Support a list of predefined jolokia agents

### ISSUE THREAD

**Issue by gashcrumb** [hawtio/hawtio-java#6]
Should definitely consider [this improvement](https://github.com/hawtio/hawtio/issues/1965) for 2.x

**Comment by pulse00** [comment]
While this feature is certainly useful, i'm wondering if there's the possibility to pass JVM agent details as URL parameters? This was possible in an earlier version of hawt.io, but has stopped working since a couple of releases ago.

**Comment by gashcrumb** [comment]
Yeah, in 2.x you can load up the console and pass a jolokia URL as a query parameter, we're using this now in a couple spots to connect to JVMs both in hawtio-kubernetes and the openshift console.  It's handled in [hawtio-jmx](https://github.com/hawtio/hawtio-jmx) in [this bit of code](https://github.com/hawtio/hawtio-jmx/blob/master/plugins/jvm/ts/jolokiaService.ts#L17-46) and supports the following query params:

`jolokiaUrl`: URL encoded string to use to connect to jolokia
`name` (optional): Name to use in the title header on the console, usually it'd be the container ID/name
`token` (optional): An oauth token, also the code checks the fragment hash for this as well
`returnTo` (optional): URL encoded string to use as a back link, used to make it easy to return back to whatever page/console the user had come from.

**Comment by pulse00** [comment]
alright, thanks. are there 2.x builds available somewhere, or do we need to build it from scratch? I could only find [1.x](http://hawt.io/getstarted/index.html) builds.

**Comment by gashcrumb** [comment]
Hmmm, looks like currently there's only a really old release of the hawtio-java repo here, hopefully we can start getting some more frequent releases out soon.  If you want to do a local build it should be a matter of just doing an `mvn install`, the hawtio-war module will contain the finished war file ready to be deployed.  One thing to note is that [this issue](https://github.com/hawtio/hawtio-java/issues/3) kinda forces you to deploy the war under /hawtio/ unless you update the base href in [this file](https://github.com/hawtio/hawtio-java/blob/master/hawtio-console-assembly/index.html).

**Comment by pulse00** [comment]
Unfortunately with the current master branch the build fails with:

```[ERROR] Failed to execute goal on project hawtio-json-schema-mbean: Could not resolve dependencies for project io.hawt:hawtio-json-schema-mbean:bundle:1.5-SNAPSHOT: The following artifacts could not be resolved: org.apache.karaf.admin:org.apache.karaf.admin.management:jar:2.4.0.redhat-620004, org.apache.karaf.features:org.apache.karaf.features.core:jar:2.4.0.redhat-620004, org.apache.karaf.features:org.apache.karaf.features.command:jar:2.4.0.redhat-620004, org.apache.karaf.shell:org.apache.karaf.shell.console:jar:2.4.0.redhat-620004, org.apache.karaf.shell:org.apache.karaf.shell.config:jar:2.4.0.redhat-620004, org.apache.karaf.shell:org.apache.karaf.shell.ssh:jar:2.4.0.redhat-620004: Failure to find org.apache.karaf.admin:org.apache.karaf.admin.management:jar:2.4.0.redhat-620004 in http://repo.jenkins-ci.org/public/ was cached in the local repository, resolution will not be reattempted until the update interval of repo.jenkins-ci.org has elapsed or updates are forced -> [Help 1]```
